### PR TITLE
Texture misc

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/backends/gl/hwmap_vaapi_gl.c [hunk 1]
```diff
@@ -82,7 +83,21 @@ static int vaapi_init(struct hwmap *hwmap, struct sxplayer_frame *frame)
         return -1;
     }
 
+    ngli_glGenTextures(gl, 1, vaapi->gl_planes);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/hwmap_vaapi_gl.c:86]: shouldn't this be 2 planes?

File: libnodegl/backends/gl/hwmap_videotoolbox_darwin_gl.c [hunk 2]
```diff
@@ -111,7 +111,7 @@ static int vt_darwin_map_plane(struct hwmap *hwmap, IOSurfaceRef surface, int in
         return -1;
     }
 
-    ngli_glBindTexture(gl, GL_TEXTURE_RECTANGLE, 0);
+    ngli_glBindTexture(gl, plane_gl->target, 0);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/hwmap_videotoolbox_darwin_gl.c:114]: Why abstract it here but not in `vt_darwin_init()`?

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]
I like this PR

File: libnodegl/backends/gl/hwmap_vaapi_gl.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/hwmap_vaapi_gl.c]: Good catch, it is fixed. For the record, playing back a single media was working fine, the special texture handle 0 was used for the chroma plane without any issue :')

File: libnodegl/backends/gl/hwmap_videotoolbox_darwin_gl.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/hwmap_videotoolbox_darwin_gl.c]: I changed the commit and went the other way around (ie: using GL_TEXTURE_RECTANGLE where relevant).

File: libnodegl/backends/gl/hwmap_vaapi_gl.c (hunk 1 above)
Comment (cboesch-gpsw) [reply libnodegl/backends/gl/hwmap_vaapi_gl.c]: Hopefully https://github.com/gopro/gopro-lib-node.gl/pull/479 will help here

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**